src/test/rpc/Subscribe_test.cpp

```diff
@@ -651,6 +651,109 @@ class Subscribe_test : public beast::unit_test::suite
 
     }
 
+    void testAccountDeliveredAmount()
+    {
+        testcase("Ledger Request, Tx Delivered Amount");
+
+        using namespace test::jtx;
+        using namespace std::chrono_literals;
+
+        Account const alice("alice");
+        Account const bob("bob");
+        Account const carol("carol");
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+
+        // number of expected 'delivered_amount' available
+        int numExpectedAvailable = 0;
+
+        Env env{*this};
+        env.fund(XRP(10000), alice, bob, carol, gw);
+        env.trust(USD(1000), alice, bob, carol);
+        env.close(NetClock::time_point{446000000s});
+
+        {
+            // add payments, but do no close until subscribed
+
+            // normal payments
+            env(pay(gw, alice, USD(50)));
+            ++numExpectedAvailable;
+            env(pay(gw, alice, XRP(50)));
+            ++numExpectedAvailable;
+
+            // partial payment
+            env(pay(gw, bob, USD(9999999)), txflags(tfPartialPayment));
+            ++numExpectedAvailable;
+            env.require(balance(bob, USD(1000)));
+
+            // failed payment
+            env(pay(gw, carol, USD(9999999)), ter(tecPATH_PARTIAL));
+            // do not incExpected - the payment failed
+            env.require(balance(carol, USD(0)));
+        }
+
+        auto wsc = makeWSClient(env.app().config());
+        Json::Value stream;
```

Comment:
`stream` scope can be narrowed.

Reply:
Fixed in `Refactor tests and resolve PR feedback 3:`